### PR TITLE
Add Image Moderation

### PR DESCRIPTION
**Pull request by maltfield**
This PR resolves this ticket:

 * https://github.com/LemmyNet/lemmy-docs/issues/293

### CODE REVIEW

File: src/users/04-moderation.md [hunk 1]
```diff
@@ -31,3 +31,229 @@ All moderation actions are taken on the context menu of posts or comments. Click
 | Purge user         | Completely delete the user, including all posts and uploaded media. Use with caution.                                                       | Admin            |
 | Purge post/comment | Completely delete the post, including attached media.                                                                                       | Admin            |
 | Appoint as admin   | Gives the user administrator status                                                                                                         | Admin            |
+
+### Image Moderation
+
+> Attribution: This documentation was originally posted by Michael Altfield [here](https://tech.michaelaltfield.net/2024/03/04/lemmy-fediverse-gdpr/)
+
+Currently there is no WUI for deleting images in Lemmy. And deleting accounts does not delete the accounts' images. For more information, please see:
```
Comment (Nutomic) [inline src/users/04-moderation.md:39]: What is wui?

File: src/users/04-moderation.md [hunk 2]
```diff
@@ -31,3 +31,229 @@ All moderation actions are taken on the context menu of posts or comments. Click
...
+X-Firefox-Spdy: h2
+user@disp9140:~$ 
+```
+
+> ⓘ Note: If you get an \``incorrect_login`\` error, then try \[a\]
+logging into the instance in your web browser and then \[b\] pasting the
+\"`https://<instance_domain>/pictrs/image/delete/<pictrs_delete_token>/<image_filename>`\"
+URL into your web browser.
+
+The image should be deleted.
```
Comment (Nutomic) [inline src/users/04-moderation.md:197]: This entire method is unnecessary as you can simply use the internal purge endpoint. So remove the section.

File: src/users/04-moderation.md [hunk 3]
```diff
@@ -31,3 +31,229 @@ All moderation actions are taken on the context menu of posts or comments. Click
...
+-   [lemmy #4433: Deleted Account should delete uploaded media
+    (pictures) too](https://github.com/LemmyNet/lemmy/issues/4433)
+-   [lemmy #4441: Users unable to delete their images (pictrs
+    API)](https://github.com/LemmyNet/lemmy/issues/4441)
+-   [lemmy #4445: Create an interface for local users to view and remove
+    images](https://github.com/LemmyNet/lemmy/issues/4445)
+-   [lemmy-ui #2359: Allow users to delete images they
+    uploaded](https://github.com/LemmyNet/lemmy-ui/issues/2359)
+-   [lemmy-ui #2360: Allow admins to view & delete uploaded
+    images](https://github.com/LemmyNet/lemmy-ui/issues/2360)
```
Comment (Nutomic) [inline src/users/04-moderation.md:50]: This issue list will very quickly get outdated as issues get closed, and no one will remember to update it. In fact some are already closed.

File: src/users/04-moderation.md [hunk 4]
```diff
@@ -31,3 +31,229 @@ All moderation actions are taken on the context menu of posts or comments. Click
 | Purge user         | Completely delete the user, including all posts and uploaded media. Use with caution.                                                       | Admin            |
 | Purge post/comment | Completely delete the post, including attached media.                                                                                       | Admin            |
 | Appoint as admin   | Gives the user administrator status                                                                                                         | Admin            |
+
+### Image Moderation
+
+> Attribution: This documentation was originally posted by Michael Altfield [here](https://tech.michaelaltfield.net/2024/03/04/lemmy-fediverse-gdpr/)
```
Comment (Nutomic) [inline src/users/04-moderation.md:37]: Remove this as attribution is handled via git log. The docs are no place to advertise your website.

Comment (maltfield) [reply src/users/04-moderation.md:37]: Sorry, CC BY-SA requires attribution.

File: src/users/04-moderation.md [hunk 5]
```diff
@@ -31,3 +31,229 @@ All moderation actions are taken on the context menu of posts or comments. Click
...
+user@host:/home/user/lemmy# 
+
+user@host:/home/user/lemmy# docker compose exec pictrs /bin/sh
+~ $ 
+```
+
+Execute the following command inside the \``pictrs`\` container.
+
+```
+wget --server-response --post-data "" --header "X-Api-Token: ${PICTRS__SERVER__API_KEY}" "http://127.0.0.1:8080/internal/purge?alias=<image_filename>"
```
Comment (Nutomic) [inline src/users/04-moderation.md:234]: You need to mention how to get the api key from docker-compose.yml

File: src/users/04-moderation.md (hunk 2 above)
Comment (maltfield) [reply src/users/04-moderation.md:197]: When I tried this on a test instance, I still got the error, which is why I included this.

File: src/users/04-moderation.md [hunk 6]
```diff
@@ -31,3 +31,229 @@ All moderation actions are taken on the context menu of posts or comments. Click
...
+SERVICE    IMAGE                            NAME
+lemmy      dessalines/lemmy:0.19.3          lemmy-lemmy-1
+lemmy-ui   dessalines/lemmy-ui:0.19.3       lemmy-lemmy-ui-1
+pictrs     docker.io/asonix/pictrs:0.5.4    lemmy-pictrs-1
+postfix    docker.io/mwader/postfix-relay   lemmy-postfix-1
+postgres   docker.io/postgres:15-alpine     lemmy-postgres-1
+proxy      docker.io/library/nginx          lemmy-proxy-1
+user@host:/home/user/lemmy# 
+
+user@host:/home/user/lemmy# docker compose exec pictrs /bin/sh
```
Comment (Nutomic) [inline src/users/04-moderation.md:227]: The name is always `pictrs`, no need for ps command.

File: src/users/04-moderation.md (hunk 3 above)
Comment (maltfield) [reply src/users/04-moderation.md:50]: > no one will remember to update it

I recommend checking your docs every release and updating them. Docs are extremely important.

File: src/users/04-moderation.md (hunk 5 above)
Comment (maltfield) [reply src/users/04-moderation.md:234]: I was unable to get a functioning instance running with `pictrs`, so I don't know what that looks like, unfortunately. Can you please post your `docker-compose.yml` file with a redacted password?

File: src/users/04-moderation.md (hunk 6 above)
Comment (maltfield) [reply src/users/04-moderation.md:227]: I think redundancy is good in documentation for clarity.

File: src/users/04-moderation.md (hunk 4 above)
Comment (Nutomic) [reply src/users/04-moderation.md:37]: This repo is under AGPL. We dont accept any other licenses for contributions.

Comment (maltfield) [reply src/users/04-moderation.md:37]: AGPL is compatible with CC BY-SA, so there is no issue re-licensing CC BY-SA content under the AGPL. And both licenses require attribution.

**Pull request closed without merging**